website/src/documentation/programming-guide.md

```diff
@@ -1970,7 +1976,1078 @@ records.apply("WriteToText",
+a row with nullable fields, where the user ensures that only one of those fields is ever set at a time. However this can
+be tedious and complex to manage. The OneOf logical type provides a value class that makes it easier to manage the type
+as a union, while still using a row with nullable fields as its underlying storage. Each logical type also has a 
+unique identifier, so they can be interpreted by other languages as well. More examples of logical types are listed 
+below.
+
+#### 6.4.1. Defining a logical type {#defining-a-logical-type}
+To define a logical type you must specify a Schema type to be used to represent the underlying type as well as a unique
+identifier for that type. A logical type imposes additional semantics on top a schema type. For example, a logical 
+type to represent nanosecond timestamps is represented as a schema containing an INT64 and an INT32 field. This schema
+alone does not say anything about how to interpret this type, however the logical type tells you that this represents
+a nanosecond timestamp, with the INT64 field representing seconds and the INT32 field representing nanoseconds.
+
+Logical types are also specified by an argument, which allows creating a class of related typed. For example, a 
+limited-precision decimal type would have an integer argument indicating how many digits of precision are represented.
+The argument is represented by a schema type, so can itself be a complex type.
+
+ {:.language-java}
+In Java, a logical type is specified as a subclass of the `LogicalType` class. A custom Java class can be specified to 
+represent the logical type and conversion functions must be supplied to convert back and forth between this Java class
+and the underlying Schema type representation. For example, the logical type representing nanosecond timestamp might
+be implemented as follows
+
+```java
+// A Logical type using java.time.Instant to represent the logical type.
+public class TimestampNanos implements LogicalType<Instant, Row> {
+  // The underlying schema used to represent rows.
+  private final Schema SCHEMA = Schema.builder().addInt64Field("seconds").addInt32Field("nanos").build();
+  @Override public String getIdentifier() { return "timestampNanos"; }
+  @Override public FieldType getBaseType() { return schema; }
+  
+  // Convert the representation type to the underlying Row type. Called by Beam when necessary.
+  @Override public Row toBaseType(Instant instant) {
+    return Row.withSchema(schema).addValues(instant.getEpochSecond(), instant.getNano()).build();
+  }
+  
+  // Convert the underlying Row type to and Instant. Called by Beam when necessary.
+  @Override public Instant toInputType(Row base) {
+    return Instant.of(row.getInt64("seconds"), row.getInt32("nanos"));
+  }
+
+     ...
+}
+```
+
+#### 6.4.2. Useful logical types {#built-in-logical-types}
+##### **EnumerationType**
+EnumerationType allows creating an enumeration type consisting of a set of named constants.
```

Comment:
`This logical type allows to create .an enum...`?

Reply:
not sure I understand?

Reply:
I think the ask is to avoid repeating EnumerationType in the description by referring to it as "This logical type"